src/CheckIfDead.php

```diff
@@ -412,27 +417,51 @@ public function sanitizeURL( $url ) {
 		if ( isset( $parts['host'] ) ) {
 			// Properly encode the host.  It can't be UTF-8.
 			// See https://en.wikipedia.org/wiki/Internationalized_domain_name.
+			// Hosts are case insensitive, might as well make them lowercase too.
 			if ( function_exists( 'idn_to_ascii' ) ) {
-				$url .= idn_to_ascii( $parts['host'] );
+				$url .= strtolower( idn_to_ascii( $parts['host'] ) );
 			} else {
-				$url .= $parts['host'];
+				$url .= strtolower( $parts['host'] );
 			}
 			if ( isset( $parts['port'] ) ) {
-				$url .= ":" . $parts['port'];
+				// Ports are only needed if not using the defaults for the scheme.
+				// Remove port numbers if they are the default.
+				switch ( $parts['port'] ) {
+					case 80:
+						if ( !isset( $parts['scheme'] ) ||
+							strtolower( $parts['scheme'] ) == "http"
+						) {
+							break;
+						}
+					case 443:
+						if ( isset( $parts['scheme'] ) &&
+							strtolower( $parts['scheme'] ) == "https"
+						) {
+							break;
+						}
+					case 21:
+						if ( isset( $parts['scheme'] ) &&
+							strtolower( $parts['scheme'] ) == "ftp"
+						) {
+							break;
+						}
+					default:
+						$url .= ":" . $parts['port'];
+				}
```

Comment:
I also don't know what 
`elseif ( isset( $parts['port'] ) ) {
	// Add port number if scheme is unknown
	$url .= ":" . $parts['port'];
}`

is supposed to be.  We can't curl a scheme less URL.